packages/dev/core/src/Meshes/Node/Blocks/linearClonerBlock.ts

```diff
@@ -0,0 +1,231 @@
+import { NodeGeometryBlock } from "../nodeGeometryBlock";
+import type { NodeGeometryConnectionPoint } from "../nodeGeometryBlockConnectionPoint";
+import { RegisterClass } from "../../../Misc/typeStore";
+import { NodeGeometryBlockConnectionPointTypes } from "../Enums/nodeGeometryConnectionPointTypes";
+import { Matrix, Quaternion, Vector3 } from "../../../Maths/math.vector";
+import { type NodeGeometryBuildState } from "../nodeGeometryBuildState";
+import type { VertexData } from "core/Meshes/mesh.vertexData";
+import { Nullable } from "core/types";
+
+/**
+ * Block used to clone geometry along a line
+ */
+export class LinearClonerBlock extends NodeGeometryBlock {
+    private _vertexData: VertexData;
+    private _currentIndex: number;
+
+    /**
+     * Gets the current index in the current flow
+     * @returns the current index
+     */
+    public getExecutionIndex(): number {
+        return this._currentIndex;
+    }
+
+    /**
+     * Gets the current face index in the current flow
+     * @returns the current face index
+     */
+    public getExecutionFaceIndex(): number {
+        return 0;
+    }
+
+    /**
+     * Create a new LinearClonerBlock
+     * @param name defines the block name
+     */
+    public constructor(name: string) {
+        super(name);
+        this.registerInput("geometry", NodeGeometryBlockConnectionPointTypes.Geometry, false);
+        this.registerInput("count", NodeGeometryBlockConnectionPointTypes.Int, true, 1, 0);
+        this.registerInput("origin", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(0, 0, 0));
+
+        //Direction is magnitude per step, or total distance
+        this.registerInput("direction", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(1, 0, 0));
+        //Per step or total
+        this.registerInput("transformPerStepOrTotal", NodeGeometryBlockConnectionPointTypes.Int, true, 0, 0, 1);

```

Comment:
sorry I'm still at that again :) I want to simplify these 2 nodes and so we should NOT have per step. It is easy for the user to provide a direction / nbsteps if they want to and that allows us to simplify this node (same for rotation and scaling)

Reply:
I think its a real determent not to have per step vs total in a cloner, they do two completely different distributions.

If its just per step as a designer if I wanted to make sure the first is at certain point and then the last is at a certain point each time I change the count you have to do finalDistance divided by count and then input that into my transform/rotation/scaling.  Then if I wanted to change the count I have to keep repeating that process vs just being able to change the count, which kind of defeats the purpose of using a linear or radial cloner.

Taking it away does not make it simpler on the user at all one could argue.

I really want to push back on removing that.

Reply:
Ok let me add @PatrickRyanMS to help convincing me :D

Reply:
I think I am falling in the middle between the two opinions. I agree that the two methods both offer a different and potentially needed utility, but I also feel that the nodes are already too complex. Adding a per-step and a total distribution option to the same node seems to complicate everything. If I were to approach the design on these nodes, I would separate the functionality into different nodes as you can simplify the inputs even more. 

**Linear Clone**
- Geometry
- Offset Position (position of final clone)
- Offset Rotation (rotation of final clone)
- Offset Scale (scale of final clone)
- Count (clones evenly distributed along interpolation between start TRS and end TRS)

**Radial Clone**
- Geometry
- Radius (radius at which clones are distributed from origin)
- Rotation (radians rotated at radius distance that the offset will travel)
- Offset Position (final position of last clone)
- Offset Scale (final scale of last clone)
- Count (clones evenly distributed along spiral)

**Repeater or Loop**
- Geometry
- Iterations (number of times to do whatever is in input)
- Input (flow of operations to repeat)

We would also need a straight clone node. The reason I would split it out this way is because the total distribution methods for the first two offer a very specific operation that allows a simple distribution method. If the user wants to get very specific about the number and offset of each clone, offering a repeater or loop node handles this. However, the repeater or loop node also opens the door to any number of operations multiple times, so it adds extra utility.

Reply:
I like the Linear Clone and the Radial Clone (that is what I wanted to say: let's start with simple. We can always add options or new blocks later on whereas if we start with the more complex we cannot simplify afterwards).

For me the Repeater or Loop are simply a Clone: generates x geometries

Reply:
Something worth noting as well to aleviate some stress from @Pryme8. I'm planning to add nodes to manipulate variables: https://github.com/BabylonJS/Babylon.js/issues/14193

Reply:
So basically separate the concerns?  Instead of two cloners that can do a mix mash, do four cloners that are Per step or Per total?

I'm down for whichever. Just want to keep it simple but effective.

Reply:
I Will do a simple clone that will handle the step version. So simply consider yours have total ===1 so we can remove the 3 stepsOrTotal parameters

Thanks buddy!!!